cpp/src/aztec/honk/proof_system/verifier.cpp

```diff
@@ -0,0 +1,121 @@
+{}
+
+template <typename program_settings>
+Verifier<program_settings>::Verifier(Verifier&& other)
+    : manifest(other.manifest)
+    , key(other.key)
+    , commitment_scheme(std::move(other.commitment_scheme))
+{}
+
+template <typename program_settings> Verifier<program_settings>& Verifier<program_settings>::operator=(Verifier&& other)
+{
+    key = other.key;
+    manifest = other.manifest;
+    commitment_scheme = (std::move(other.commitment_scheme));
+    kate_g1_elements.clear();
+    kate_fr_elements.clear();
+    return *this;
+}
+
+/**
+* @brief This function verifies a Honk proof for given program settings.
+*
+* TODO(luke): Complete this description
+* @detail A Standard Honk proof contains the following:
+    Multilinear evaluations:
+        w_i(X),        i = 1,2,3
+        sigma_i(X),    i = 1,2,3
+        q_i(X),        i = 1,2,3,4,5
+        z_perm(X),
+        L_0(X),
+        id(X)
+
+    Univariate evaluations:
+        a_0 = Fold_{-r}^(0)(-r),
+        a_l = Fold^(l)(-r^{2^l}), i = 1,...,d-1
+
+    Univariate polynomials (evaluations over MAX_RELATION_LENGTH-many points):
+        S_l, l = 0,...,d-1
+
+    Commitments:
+        [w_i]_1,        i = 1,2,3
+        [z_perm]_1,
+        [Fold^(l)]_1,   l = 1,...,d-1
+        [Q]_1,
+        [W]_1
+*/
+template <typename program_settings> bool Verifier<program_settings>::verify_proof(const waffle::plonk_proof& proof)
+{
+    // TODO(luke): TBD how 'd' gets set here and elsewhere
+    const size_t multivariate_d(1);
+
+    const size_t num_polys = program_settings::num_polys;
+    using FF = typename program_settings::fr;
+    using Transcript = typename program_settings::Transcript;
+    using Multivariates = Multivariates<FF, num_polys, multivariate_d>;
+
+    key->program_width = program_settings::program_width;
+
+    // Add the proof data to the transcript, according to the manifest. Also initialise the transcript's hash type
+    // and challenge bytes.
+    auto transcript = transcript::StandardTranscript(
+        proof.proof_data, manifest, program_settings::hash_type, program_settings::num_challenge_bytes);
+
+    // Add the circuit size and the number of public inputs) to the transcript.
+    transcript.add_element("circuit_size",
+                           { static_cast<uint8_t>(key->n >> 24),
+                             static_cast<uint8_t>(key->n >> 16),
+                             static_cast<uint8_t>(key->n >> 8),
+                             static_cast<uint8_t>(key->n) });
+
+    transcript.add_element("public_input_size",
+                           { static_cast<uint8_t>(key->num_public_inputs >> 24),
+                             static_cast<uint8_t>(key->num_public_inputs >> 16),
+                             static_cast<uint8_t>(key->num_public_inputs >> 8),
+                             static_cast<uint8_t>(key->num_public_inputs) });
+
+    // Compute challenges from the proof data, based on the manifest, using the Fiat-Shamir heuristic
```

Comment:
Don't we have to get more challenges than this? E.g., we need to generate `u` and then compute the values of some polynomials there.

Reply:
Yeah but they have to exist in the manifest. To me it made sense to just wait for that to be done rather than duplicating your manifest work by mocking it

Reply:
Improved comment for future TODO on this